### ISSUE TITLE
protobuf-src compile time increased massively from v1.1.0 -> v2.1.0

### ISSUE THREAD

**Issue by paulheinr** [MaterializeInc/rust-protobuf-native#24]
Hi there, this is more a discussion than an issue. We are using protobuf-src in a project: https://github.com/matsim-vsp/parallel_qsim_rust

I wanted to switch from v1.1.0 to v2.1.0 and noticed an increased compile time of `protobuf-src` (from 1-2min to 6-7min) on my Mac. I am wondering whether any body else is facing the same issues. I am considering a downgrade because the compile time is too long. 

Do you have an idea why this is the case and how to accelerate the compile process?  

### System settings
- Processor: 2,6 GHz 6-Core Intel Core i7
- Memory: 32 GB 2667 MHz DDR4
- OS: macOS 14.5
- cmake version 3.31.0
- cargo 1.82.0
- rustc 1.82.0

**Comment by antiguru** [comment]
This is your problem: rustc 1.82.0. See https://github.com/rust-lang/rust/issues/132064

**Comment by paulheinr** [comment]
Thank's for the hint!

**Comment by kesyog** [comment]
FYI there's a great chance this issue would be fixed by https://github.com/rust-lang/cmake-rs/pull/229. One workaround in the meantime is to set the `CMAKE_BUILD_PARALLEL_LEVEL` environment variable based on `NUM_JOBS` for OSX builds.

**Comment by teskje** [comment]
I think this isn't caused by the rustc 1.82.0 slowness after all. With rustc 1.83.0 I'm seeing the same thing on my Macbook:

```
$ cd protobuf-src
$ git checkout protobuf-src-v1.1.0git checkout protobuf-src-v1.1.0
$ cargo clean
$ cargo build
   Compiling cc v1.0.72
   Compiling autotools v0.2.5
   Compiling protobuf-src v1.1.0+21.5 (/Users/jan/devel/rust-protobuf-native/protobuf-src)
    Finished `dev` profile [unoptimized + debuginfo] target(s) in 47.80s
$ git checkout protobuf-src-v2.0.0
$ cargo clean
$ cargo build
   Compiling cc v1.0.97
   Compiling cmake v0.1.50
   Compiling protobuf-src v2.0.0+26.1 (/Users/jan/devel/rust-protobuf-native/protobuf-src)
    Finished `dev` profile [unoptimized + debuginfo] target(s) in 4m 53s
```

Seeing how cmake only pops up with v2.0.0, @kesyog's diagnosis seems right.

**Comment by teskje** [comment]
```
$ git checkout protobuf-src-v2.0.0
$ cargo clean
$ env CMAKE_BUILD_PARALLEL_LEVEL=10 cargo build
   Compiling cc v1.0.97
   Compiling cmake v0.1.50
   Compiling protobuf-src v2.0.0+26.1 (/Users/jan/devel/rust-protobuf-native/protobuf-src)
    Finished `dev` profile [unoptimized + debuginfo] target(s) in 53.77s
```

See also the discussion in https://github.com/MaterializeInc/rust-protobuf-native/pull/25

**Comment by benesch** [comment]
Yeah, I think this is almost certainly what @kesyog reported. Still waiting on his PR to cmake-rs to be merged and released. 😞

**Comment by kesyog** [comment]
My PR got merged and the changes should be available in cmake-rs 0.1.53
